domain/product/entity/entity.go

```diff
@@ -45,6 +45,22 @@ type Product struct {
 	ProductTypeName     string          `gorm:"-"`
 	ProductCategoryID   int64           `gorm:"-"`
 	ProductCategoryName string          `gorm:"-"`
+	Categories          []*Category     `gorm:"many2many:product_category_relations;"`
+}
+
+type Category struct {
+	base_model.BaseMasterDataModel
+	Name     string    `gorm:"type:varchar(255);not null;unique"`
+	ParentID int64     `gorm:"type:int;null"`
+	IsActive bool      `gorm:"type:bool;not null"`
+	Products []Product `gorm:"many2many:product_category_relations;"`
+}
+
+type ProductCategoryRelation struct {
```

Comment:
for the **product category relation** table, do we need to add `id `and set it as Primary key mas?

Reply:
based on my experience, we don't really need primary_id in the relation table since it is used for reference only, no specific operation directly to this table. Additionally, the combination of the product_id and category_id is already unique, if the data whether product or category is removed, we expect that this relation data should be removed as well.

or do you have other concerns regarding this mas?

Reply:
Please take a look for the other tables in the database that have the same behaviour with this (new table because many to many relationship).
I have no concern mas regarding this, just to make sure it's necessary or not.

Reply:
as many samples of relation tables that act as junctions don't need to have it's own ID, hence I believe with only product_id and category_id is fine for this case, as both of the ID is already an index.